expression/builtin_like_vec.go

```diff
@@ -27,17 +30,83 @@ func (b *builtinLikeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) er
 }
 
 func (b *builtinRegexpBinarySig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpBinarySig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	return b.builtinRegexpSharedSig.vecEvalInt(input, result, b)
 }
 
 func (b *builtinRegexpSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinRegexpSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	return b.builtinRegexpSharedSig.vecEvalInt(input, result, b)
+}
+
+func (b *builtinRegexpSharedSig) isMemoizedRegexpInitialized() bool {
+	return !(b.memoizedRegexp == nil && b.memoizeErr == nil)
+}
+
+func (b *builtinRegexpSharedSig) initMemoizedRegexp(patterns *chunk.Column, n int, rc regexpCompiler) {
+	// Precondition: patterns is generated from a constant expression
+	for i := 0; i < n; i++ {
+		if patterns.IsNull(i) {
+			continue
+		}
+		re, err := rc.compile(patterns.GetString(i))
+		b.memoizedRegexp = re
+		b.memoizeErr = err
+		break
+	}
+	if !b.isMemoizedRegexpInitialized() {
+		b.memoizeErr = errors.New("No valid regexp pattern found")
+	}
+}
+
+func (b *builtinRegexpSharedSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column, rc regexpCompiler) error {
+	n := input.NumRows()
+	bufExpr, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufExpr)
+	if err := b.args[0].VecEvalString(b.ctx, input, bufExpr); err != nil {
+		return err
+	}
+
+	bufPat, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(bufPat)
+	if err := b.args[1].VecEvalString(b.ctx, input, bufPat); err != nil {
+		return err
+	}
+
+	if b.args[1].ConstItem() && !b.isMemoizedRegexpInitialized() {
+		b.initMemoizedRegexp(bufPat, n, rc)
+	}
+	getRegexp := func(pat string) (*regexp.Regexp, error) {
+		if b.memoizedRegexp != nil {
+			return b.memoizedRegexp, nil
+		}
```

Comment:
Should we check `memoizeErr` here?

Reply:
SG. I updated `initMemoizedRegexp()` so that if there is an error during the initialization, `memoizedRegexp` will be set to `nil`.

Reply:
If an error happens, why shouldn't we just return the `memoizeErr` in this function for fast path? But continue to do `rc.compile(pat)` and get a same error?

Reply:
I see your point, fixed